tests/dashboards/security/security_dataset_tests.py

```diff
@@ -0,0 +1,260 @@
+        }
+
+        # act
+        responses_by_url = {
+            url: self.client.get(url).data.decode("utf-8")
+            for url in dashboard_title_by_url.keys()
+        }
+
+        # assert
+        for dashboard_url, get_dashboard_response in responses_by_url.items():
+            assert (
+                escape(dashboard_title_by_url[dashboard_url]) in get_dashboard_response
+            )
+
+    def test_get_dashboards__users_are_dashboards_owners(self):
+        # arrange
+        username = "gamma"
+        user = security_manager.find_user(username)
+        my_owned_dashboard = create_dashboard_to_db(
+            dashboard_title="My Dashboard",
+            slug=f"my_dash_{random_slug()}",
+            published=False,
+            owners=[user],
+        )
+
+        not_my_owned_dashboard = create_dashboard_to_db(
+            dashboard_title="Not My Dashboard",
+            slug=f"not_my_dash_{random_slug()}",
+            published=False,
+        )
+
+        self.login(user.username)
+
+        # act
+        get_dashboards_response = self.get_resp(DASHBOARDS_API_URL)
+
+        # assert
+        self.assertIn(my_owned_dashboard.url, get_dashboards_response)
+        self.assertNotIn(not_my_owned_dashboard.url, get_dashboards_response)
+
+    def test_get_dashboards__owners_can_view_empty_dashboard(self):
+        # arrange
+        dash = create_dashboard_to_db("Empty Dashboard", slug="empty_dashboard")
+        dashboard_url = dash.url
+        gamma_user = security_manager.find_user("gamma")
+        self.login(gamma_user.username)
+
+        # act
+        get_dashboards_response = self.get_resp(DASHBOARDS_API_URL)
+
+        # assert
+        self.assertNotIn(dashboard_url, get_dashboards_response)
+
+    def test_get_dashboards__users_can_view_favorites_dashboards(self):
+        # arrange
+        user = security_manager.find_user("gamma")
+        fav_dash_slug = f"my_favorite_dash_{random_slug()}"
+        regular_dash_slug = f"regular_dash_{random_slug()}"
+
+        favorite_dash = Dashboard()
+        favorite_dash.dashboard_title = "My Favorite Dashboard"
+        favorite_dash.slug = fav_dash_slug
+
+        regular_dash = Dashboard()
+        regular_dash.dashboard_title = "A Plain Ol Dashboard"
+        regular_dash.slug = regular_dash_slug
+
+        db.session.merge(favorite_dash)
+        db.session.merge(regular_dash)
+        db.session.commit()
+
+        dash = db.session.query(Dashboard).filter_by(slug=fav_dash_slug).first()
+
+        favorites = models.FavStar()
+        favorites.obj_id = dash.id
+        favorites.class_name = "Dashboard"
+        favorites.user_id = user.id
+
+        db.session.merge(favorites)
+        db.session.commit()
+
+        self.login(user.username)
+
+        # act
+        get_dashboards_response = self.get_resp(DASHBOARDS_API_URL)
+
+        # assert
+        self.assertIn(f"/superset/dashboard/{fav_dash_slug}/", get_dashboards_response)
```

Comment:
What's the benefit of testing favorited dashboards separately? Do we expect a risk that favorites somehow wouldn't be accessible?

Reply:
actually for a reason which is unclear to me access to fav dashboards is managed just by setting the fav.
even without any data/role access
https://github.com/apache/superset/blob/f6c7cbeeb40a3f1400ecc8e1b6b371b008f67397/superset/dashboards/filters.py#L106

is it a mistake?
bc we already have the 
https://github.com/apache/superset/blob/f6c7cbeeb40a3f1400ecc8e1b6b371b008f67397/superset/dashboards/filters.py#L106
which handles filters favs on top of access

Reply:
This is a good point, might be some legacy logic, removing it is a breaking so it get perpetuated. Maybe @betodealmeida can help with more context?

Reply:
move to [discussion](https://github.com/apache/superset/discussions/12839)